client/state/ui/editor/image-editor/reducer.js

```diff
@@ -59,7 +60,12 @@ export function hasChanges( state = false, action ) {
 export function fileInfo( state = defaultFileInfo, action ) {
 	switch ( action.type ) {
 		case IMAGE_EDITOR_SET_FILE_INFO:
-			return Object.assign( {}, state, { src: action.src, fileName: action.fileName, mimeType: action.mimeType } );
+			return Object.assign( {}, state, {
```

Comment:
Updated in ff07515. I also fixed a case where the title wouldn't update if the original was attached to any post.